code/drasil-code/lib/Language/Drasil/Code/Imperative/Import.hs

```diff
@@ -354,6 +363,10 @@ convExpr (Matrix [l]) = do
                                     -- hd will never fail here
   return $ litArray (fmap valueType (head ar)) ar
 convExpr Matrix{} = error "convExpr: Matrix"
+convExpr (S.Set l) = do
```

Comment:
It could be added to `Set`s I'm just not sure if it is necessary

Reply:
I'm not sure either... but doing (late) type reconstruction is usually a bad idea.

Reply:
Is there any other way to do the type reconstruction other then giving a space to set?

Reply:
Sorry, I was too obtuse: you shouldn't be doing 'type reconstruction' at all. You should know, right when the set is created, what kind of thing it is going to hold. You shouldn't be asking the "values" in the set what type they are, because you should already know it.

Reply:
If I change `set` to 'Set :: Space -> [Expr] ->Expr' I can use the `Space` to store the type of the set

Reply:
Yes, that is what I was suggesting.